### PR TITLE
cnosdb 1.0.2 (new formula)

### PR DESCRIPTION
**Pull request by xmh1011**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
Continuation of https://github.com/Homebrew/homebrew-core/pull/103604

**Comment by SMillerDev** [comment]
Please don't create new pull requests once you change something. It makes it really hard to keep track of comments.

**Comment by xmh1011** [comment]
> Please don't create new pull requests once you change something. It makes it really hard to keep track of comments.

Thank you very much for your help and I'm sorry to trouble you all the time. I really don't know enough about Ruby scripts, sorry.

**Comment by xmh1011** [comment]
> Continuation of #103604

**Comment by xmh1011** [comment]
> Continuation of #103604

do you think this pull request could be merged? Thank you.

File: Formula/cnosdb.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      s.gsub! %r{/.*/.cnosdb/meta}, "#{testpath}/cnosdb/meta"
+      s.gsub! %r{/.*/.cnosdb/wal}, "#{testpath}/cnosdb/wal"
+    end
+    begin
+      cnosdb = fork do
+        exec "#{bin}/cnosdb-cli", "--host=#{cnosdb_host}", "--port=#{cnosdb_port}"
+      end
+      sleep 30
+      # Assert that initial resources show in CLI output.
+      assert_match "CnosDB v1.0.2 (git: unknown bf128f1533)", shell_output("#{bin}/cnosdb version")
```
Comment (SMillerDev) [inline Formula/cnosdb.rb:59]:
```suggestion
      assert_match "CnosDB v#{version}", shell_output("#{bin}/cnosdb version")
```

File: Formula/cnosdb.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      s.gsub! "/var/lib/cnosdb/data", "#{var}/cnosdb/data"
+      s.gsub! "/var/lib/cnosdb/meta", "#{var}/cnosdb/meta"
+      s.gsub! "/var/lib/cnosdb/wal", "#{var}/cnosdb/wal"
+    end
+    etc.install "#{HOMEBREW_PREFIX}/etc/config.sample.toml" => "cnosdb.conf"
+    (var/"cnosdb/data").mkpath
+    (var/"cnosdb/meta").mkpath
+    (var/"cnosdb/wal").mkpath
+  end
+  service do
```
Comment (SMillerDev) [inline Formula/cnosdb.rb:37]:
```suggestion
  end

  service do
```

File: Formula/cnosdb.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      revision: "bf128f15334a9104a5a644917693c29bf5eb3e65"
+  license "MIT"
+  head "https://github.com/cnosdb/cnosdb.git", branch: "main"
+  # The regex below omits a rogue `v9.9.9` tag that breaks version comparison.
+  livecheck do
+    url :stable
+    regex(/^v?((?!9\.9\.9)\d+(?:\.\d+)+)$/i)
+  end
+  depends_on "go" => :build
+  def install
```
Comment (SMillerDev) [inline Formula/cnosdb.rb:15]:
```suggestion
  head "https://github.com/cnosdb/cnosdb.git", branch: "main"

  # The regex below omits a rogue `v9.9.9` tag that breaks version comparison.
  livecheck do
    url :stable
    regex(/^v?((?!9\.9\.9)\d+(?:\.\d+)+)$/i)
  end

  depends_on "go" => :build

  def install
```

**Comment by xmh1011** [comment]
Can this pull request be merged? Thank you, sir. @SMillerDev

**Comment by SMillerDev** [comment]
Not while CI is failing.

**Comment by xmh1011** [comment]
The test failed with the message "Error: Formula was not installed with --build-bottle: cnosdb".  Can you tell me how to fix the error.  I have tried my best and googled, but I don't know the solution to it. Could you teach me that? Thank you, sir. You have helped me a lot. If this error fixed, this pull request can be merged. Thank you!

**Pull request closed without merging**